dotcom-rendering/src/web/lib/notification.ts

```diff
@@ -52,31 +44,20 @@ export const addNotificationsToDropdownLinks = (
 	const notificationsByTarget = groupNotificationsByTarget(notifications);
 
 	const linksWithNotifications = links.map((link) => {
-		const targetHasNewNotifications = Object.prototype.hasOwnProperty.call(
-			notificationsByTarget,
-			link.id,
```

Comment:
Checking if something exists before accessing it creates a race condition. Here we try accessing it directly, and default to an empty array.

Reply:
Is that true in JavaScript browser envs given [run to completion](https://developer.mozilla.org/en-US/docs/Web/JavaScript/EventLoop#run-to-completion) guarantees (assuming we're not using generators which I think muddies the water somewhat). (Purely asking because I'm interested, the code looks good!)

Reply:
Ah yes, that's a good point, but I think that the code also happens to be easier to reason about!